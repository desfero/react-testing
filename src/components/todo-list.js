import React from 'react';
import PropTypes from 'prop-types';
import {List} from 'immutable';

import {TodoItem} from './todo-item';

export const TodoList = ({todos, removeTodo, changeTodoState}) => (
    <ul className="list-group list-group-flush">
        {todos.map(todo => (
            <li key={todo.id} className="list-group-item"><TodoItem item={todo} removeTodo={removeTodo} changeTodoState={changeTodoState}/></li>
        ))}
    </ul>
);

TodoList.propTypes = {
    removeTodo: PropTypes.func.isRequired,
    todos: PropTypes.instanceOf(List).isRequired,
};