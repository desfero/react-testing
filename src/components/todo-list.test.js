import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import { TodoList } from './todo-list';

describe('todo-list', () => {
    const todos = [
        {id: 1, title: 'todo 1', completed: false},
        {id: 2, title: 'todo 2', completed: true},
        {id: 3, title: 'todo 3', completed: false}
    ];

    it('should render component', () => {
        const removeTodo = () => {};
        const wrapper = shallow(<TodoList removeTodo={removeTodo} todos={new List(todos)}/>);

        expect(wrapper).toMatchSnapshot();
    });
});