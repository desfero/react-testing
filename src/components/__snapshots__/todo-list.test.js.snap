// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render the component 1`] = `
<ul
  className="list-group list-group-flush"
>
  <li
    className="list-group-item"
    key="1"
  >
    <Component
      completeTodo={[Function]}
      item={
        Object {
          "done": false,
          "id": 1,
          "title": "Something",
        }
      }
      removeTodo={[Function]}
    />
  </li>
</ul>
`;
